### PR TITLE
Change the ClusterStatusConditionType "Failing" to "Degraded" 

### PR DESCRIPTION
**Pull request by LorbusChris**
**- What I did**
- Bump vendored openshift/api and openshift/client-go
- Rename ClusterStatusConditionType "Failing" to "Degraded" 

**- How to verify it**
CI

**- Description for the changelog**
Closes: #641

### CODE REVIEW

File: pkg/apis/machineconfiguration.openshift.io/v1/helpers.go [hunk 1]
```diff
@@ -213,11 +213,11 @@ func IsControllerConfigCompleted(ccName string, ccGetter func(string) (*Controll
...
-	failing := IsControllerConfigStatusConditionTrue(cur.Status.Conditions, TemplateContollerFailing)
+	degraded := IsControllerConfigStatusConditionTrue(cur.Status.Conditions, TemplateContollerDegraded)
 	if completed &&
 		!running &&
-		!failing {
+		!degraded {
 		return nil
 	}
-	return fmt.Errorf("ControllerConfig has not completed: completed(%v) running(%v) failing(%v)", completed, running, failing)
+	return fmt.Errorf("ControllerConfig has not completed: completed(%v) running(%v) degraded(%v)", completed, running, degraded)
```
Comment (runcom) [inline pkg/apis/machineconfiguration.openshift.io/v1/helpers.go]: This is the ControllerConfig not the Operator Status, this must stay Failing

File: pkg/apis/machineconfiguration.openshift.io/v1/helpers_test.go [hunk 2]
```diff
@@ -127,7 +127,7 @@ func TestIsControllerConfigCompleted(t *testing.T) {
 		obsrvdGen int64
 		completed bool
 		running   bool
-		failing   bool
+		degraded  bool
```
Comment (runcom) [inline pkg/apis/machineconfiguration.openshift.io/v1/helpers_test.go]: back to failing this is not the Operator

File: pkg/apis/machineconfiguration.openshift.io/v1/types.go [hunk 3]
```diff
@@ -167,7 +167,7 @@ type ControllerConfigStatusConditionType string
 const (
 	TemplateContollerRunning   ControllerConfigStatusConditionType = "TemplateContollerRunning"
 	TemplateContollerCompleted ControllerConfigStatusConditionType = "TemplateContollerCompleted"
-	TemplateContollerFailing   ControllerConfigStatusConditionType = "TemplateContollerFailing"
+	TemplateContollerDegraded  ControllerConfigStatusConditionType = "TemplateContollerDegraded"
```
Comment (runcom) [inline pkg/apis/machineconfiguration.openshift.io/v1/types.go]: still Failing

File: pkg/controller/template/status.go [hunk 4]
```diff
@@ -14,7 +14,7 @@ import (
 
 // - sets `running` condition to `true`.
 // - reset the `available` condition to `false` when we are syncing to new generation.
-// - does not modify `failing` condition.
+// - does not modify `degraded` condition.
```
Comment (runcom) [inline pkg/controller/template/status.go]: all of this file is still failing as well

File: pkg/controller/template/template_controller.go [hunk 5]
```diff
@@ -410,23 +410,23 @@ func (ctrl *Controller) syncControllerConfig(key string) error {
 	if cfg.Spec.PullSecret != nil {
 		secret, err := ctrl.kubeClient.CoreV1().Secrets(cfg.Spec.PullSecret.Namespace).Get(cfg.Spec.PullSecret.Name, metav1.GetOptions{})
 		if err != nil {
-			return ctrl.syncFailingStatus(cfg, err)
+			return ctrl.syncDegradedStatus(cfg, err)
```
Comment (runcom) [inline pkg/controller/template/template_controller.go]: failing

File: pkg/controller/template/template_controller_test.go [hunk 6]
```diff
@@ -280,7 +280,7 @@ func TestCreatesMachineConfigs(t *testing.T) {
 	ccc.Status.Conditions = []mcfgv1.ControllerConfigStatusCondition{
 		{Type: mcfgv1.TemplateContollerCompleted, Status: corev1.ConditionTrue, Reason: "sync completed towards (1) generation using controller version 0.0.0-was-not-built-properly"},
 		{Type: mcfgv1.TemplateContollerRunning, Status: corev1.ConditionFalse},
-		{Type: mcfgv1.TemplateContollerFailing, Status: corev1.ConditionFalse},
+		{Type: mcfgv1.TemplateContollerDegraded, Status: corev1.ConditionFalse},
```
Comment (runcom) [inline pkg/controller/template/template_controller_test.go]: failing

File: pkg/operator/status.go [hunk 7]
```diff
@@ -28,7 +28,7 @@ func (optr *Operator) syncVersion() error {
 	}
 
 	// keep the old version and progressing if we fail progressing
-	if cov1helpers.IsStatusConditionTrue(co.Status.Conditions, configv1.OperatorProgressing) && cov1helpers.IsStatusConditionTrue(co.Status.Conditions, configv1.OperatorFailing) {
+	if cov1helpers.IsStatusConditionTrue(co.Status.Conditions, configv1.OperatorProgressing) && cov1helpers.IsStatusConditionTrue(co.Status.Conditions, configv1.OperatorDegraded) {
```
Comment (runcom) [inline pkg/operator/status.go]: right, this is correct, we're changing the Operator's StatusCondition from Failing to Degraded (the changes above aren't about the Operator)

**Comment by LorbusChris** [comment]
oops :man_facepalming: fixed

**Pull request merged**